corehq/messaging/scheduling/views.py

```diff
@@ -1070,24 +1072,28 @@ def post(self, request, *args, **kwargs):
                     deleted=False,
                 )
             except AutomaticUpdateRule.DoesNotExist:
-                errors.append(_("Row {index}, with rule id {id}").format(index=index, id=row['id']))
+                messages.error(request, _("Could not find rule for row {index}, with id {id}.").format(
+                    index=index, id=row['id']))
             dirty = False
             if rule:
-                if rule.name != row['name']:
-                    dirty = True
-                    rule.name = row['name']
-                if rule.case_type != row['case_type']:
-                    dirty = True
-                    rule.case_type = row['case_type']
+                if not isinstance(_get_rule_content(rule), SMSContent):
+                    messages.error(request, _("Row {index}, with rule id {id}, does not use SMS content.").format(
+                        index=index, id=row['id']))
+                else:
+                    if rule.name != row['name']:
+                        dirty = True
+                        rule.name = row['name']
+                    if rule.case_type != row['case_type']:
+                        dirty = True
+                        rule.case_type = row['case_type']
             if dirty:
                 rule.save()
                 success_count += 1
 
-        if errors:
-            messages.error(request, _("Could not find rules from the following rows: {errors}").format(
-                errors=", ".join(errors)))
-
-        if success_count:
-            messages.success(request, _("Updated {count} rule(s)").format(count=success_count))
+        messages.success(request, _("Updated {count} rule(s).").format(count=success_count))
```

Comment:
Assuming it's not an issue to call `message.success(...)` after `messages.error(...)`, and errors will still be displayed.

Reply:
Yep, you can add any number of messages, in whatever order and whatever type (error, success, warning, etc), and they'll all get displayed.